modules/drivers/sqlserver/src/metabase/driver/sqlserver.clj

```diff
@@ -271,6 +277,97 @@
         (hx/->AtTimeZone (zone-id->windows-zone target-timezone))
         hx/->datetime)))
 
+(defmethod sql.qp/->honeysql [:sqlserver :datetime-diff]
+  [driver [_ x y unit]]
+  (let [x (sql.qp/->honeysql driver x)
+        y (sql.qp/->honeysql driver y)
+        disallowed-types (keep
+                          (fn [v]
+                            (some-> v
+                                    hx/type-info
+                                    hx/type-info->db-type
+                                    name
+                                    str/lower-case
+                                    #{"time"}))
+                          [x y])
+        x (if (hx/is-of-type? x "datetimeoffset")
+            (hx/->AtTimeZone x (zone-id->windows-zone (qp.timezone/results-timezone-id)))
+            x)
+        x (hx/cast "datetime2" x)
+        y (if (hx/is-of-type? y "datetimeoffset")
+            (hx/->AtTimeZone y (zone-id->windows-zone (qp.timezone/results-timezone-id)))
+            y)
+        y (hx/cast "datetime2" y)]
+    (when (seq disallowed-types)
+      (throw (ex-info (tru "Only datetime, timestamp, or date types allowed. Found {0}"
+                           (pr-str disallowed-types))
+                      {:found disallowed-types
+                       :type  qp.error-type/invalid-query})))
```

Comment:
```suggestion
                       :type  qp.error-type/invalid-parameter})))
```
seems more like a `parameter issue`?

Reply:
I've had this discussion with cam before. `qp.error-type/invalid-parameter` is reserved for sql parameters as in https://www.metabase.com/docs/latest/questions/native-editor/sql-parameters